scripts/gen-edpm-kustomize.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+#
+# Copyright 2023 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+set -ex
+
+# expect that the common.sh is in the same dir as the calling script
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+. ${SCRIPTPATH}/common.sh --source-only
+
+if [ -z "$NAMESPACE" ]; then
+    echo "Please set NAMESPACE"; exit 1
+fi
+
+if [ -z "$KIND" ]; then
+    echo "Please set SERVICE"; exit 1
+fi
+
+if [ -z "$DEPLOY_DIR" ]; then
+    echo "Please set DEPLOY_DIR"; exit 1
+fi
+
+NAME=${KIND,,}
+
+if [ ! -d ${DEPLOY_DIR} ]; then
+    mkdir -p ${DEPLOY_DIR}
+fi
+
+pushd ${DEPLOY_DIR}
+
+cat <<EOF >kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+namespace: ${NAMESPACE}
+patches:
+- target:
+    kind: ${KIND}
+  patch: |-
+    - op: replace
+      path: /spec/deployStrategy/deploy
+      value: true
+    - op: replace
+      path: /spec/nodes/edpm-compute-0/ansibleHost
+      value: ${EDPM_COMPUTE_IP}
+    - op: replace
+      path: /spec/nodes/edpm-compute-0/openStackAnsibleEERunnerImage
+      value: ${OPENSTACK_RUNNER_IMG}
+    - op: replace
+      path: /spec/nodes/edpm-compute-0/node/ansibleSSHPrivateKeySecret
+      value: ${EDPM_ANSIBLE_SECRET}
+    - op: replace
+      path: /spec/nodes/edpm-compute-0/node/ansibleVars
+      value: |
+        ctlplane_ip: ${EDPM_COMPUTE_IP}
```

Comment:
@gibizer I have one stupid query, Do we need to keep EDPM host and edpm control plane ip same?
Reason I am asking because in Zuul job, we have external node with public ip. Before [network isolation](https://github.com/openstack-k8s-operators/install_yamls/commit/06b8f5e3c3e381a82a3f567621d0d52f90a3bc89#diff-c5f64cab8f76db38e0f0d1e3961e32ead7281c4328598f7dcca0fdf9f9051d8cL65)
I tried with different ips( _COMPUTE_IP as public ip and using the default one as ctlplane ip) and it works fine. After network isolation patch merged, I work around that issue by replace ctlplane ip with[ previous one](https://github.com/rdo-infra/rdo-jobs/blob/master/playbooks/edpm/workarounds.yaml#L2).

If it does not needs to be same, can we also parameterized the ctlplane_ip ?

Reply:
@raukadah I'm not sure. @fao89 @stuggi could you help with this?

Reply:
I just tried to keep the same structure from edpm_play: https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/edpm/edpm-play.yaml#L50-L56

Reply:
@raukadah sorry I don't getting the network arch when using zuul job. do we use dedicated nodes for OCP and edpm, or is it also on one using VMs for OCP and edpm node? what you mean with `default one`? 

@fao89 right now my understanding was that the EDPM host and edpm control plane ip is the same (provisioning network), but maybe I am wrong? If it isn't, could we just remove the ctlplane network at all?

In general the deployment pods, triggered by the OpenStackDataPlaneNode, and running on OCP need to be able to reach the edpm node via ssh. I realized right now we don't attach the job pods direct to the ctlplane network, which works on an env where the the deployment pods can reach the nodes via the pod network. For environments where this is not possible you'd have to add the correct network via spec.networkAttachments, like e.g.
```
  networkAttachments:
  - ctlplane
```

Reply:
I believe they are the same, but I'm not sure, @slagle could you please confirm?

Reply:
Yes, Host (which I assume you mean 192.168.122.x IP) and ctlplane_ip need to be the same. That is due to the edpm_network_config_template which is being used. That is configured to be:

```
DATAPLANE_NETWORK_CONFIG_TEMPLATE  ?=templates/single_nic_vlans/single_nic_vlans.j2
```

If you look at the single_nic_vlans.j2 template from edpm-ansible, https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_network_config/templates/single_nic_vlans/single_nic_vlans.j2, then the ctlplane network uses nic1, which is eth0, which has the host ip.  That gets added to the ovs bridge.

Reply:
@stuggi @gibizer @slagle @fao89  thank you everyone for the reply. In Zuul CI, the crc vm is running inside nodepool node. Here is the network output of the [nodepool node](https://logserver.rdoproject.org/openstack-periodic-container-zed-centos9/review.rdoproject.org/rdo-jobs/master/periodic-centos-9-podified-edpm-integration-zed/ca13786/controller/controller/network.txt). 

The EDPM node is another nodepool node and having following [network config](https://logserver.rdoproject.org/openstack-periodic-container-zed-centos9/review.rdoproject.org/rdo-jobs/master/periodic-centos-9-podified-edpm-integration-zed/ca13786/secondary/secondary/network.txt).

I am using [multinode.j2](https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_network_config/templates/ci/multinode.j2) as EDPM_NETWORK_CONFIG_TEMPLATE. Here is the oc describe output of[ EDPM play](https://logserver.rdoproject.org/openstack-periodic-container-zed-centos9/review.rdoproject.org/rdo-jobs/master/periodic-centos-9-podified-edpm-integration-zed/ca13786/controller/controller/pod/deploy-external-dataplane-compute-5wv6p-describe.txt). 
```
                                            38.102.83.87:
                                                     ansible_ssh_user: root
                                                     ctlplane_ip: 192.168.24.100
                                                     internal_api_ip: 172.17.0.100
                                                     storage_ip: 172.18.0.100
                                                     tenant_ip: 172.10.0.100
                                                     fqdn_internal_api: '{{ ansible_fqdn }}'
```
deploy-external-dataplane-compute pod is able to connect to external compute and run ansible [plays ](https://logserver.rdoproject.org/openstack-periodic-container-zed-centos9/review.rdoproject.org/rdo-jobs/master/periodic-centos-9-podified-edpm-integration-zed/ca13786/controller/controller/pod/deploy-external-dataplane-compute-5wv6p-logs.txt)on that.

This way I was testing edpm-ansible and openstack-ansiblee-operator changes.
I was confused about ctlplane_ip starting with 192.168.24.x. that's why I have asked the question.
The default network config template was not working for me and we lose connectivity to external compute (the same issue was faced with traditional edpm job). So I was using multinode.j2 template and it was working fine.
I also tried with single_nic_vlans.j2 but network connectivity (ssh + ping) both gets lost from that.

I am not sure which network_config_template will work for CI case. That's why I am asking these weired questions.